funding/manager.go

```diff
@@ -4497,6 +4497,259 @@ func (f *Manager) newChanAnnouncement(localPubKey,
+	if err != nil && !errors.Is(err, channeldb.ErrChannelNotFound) {
+		return nil, errors.Errorf("unable to generate channel "+
+			"update announcement: %v", err)
+	}
+
+	switch {
+	case ourPolicy != nil:
+		// If ourPolicy is non-nil, modify the default parameters of the
+		// ChannelUpdate.
+		chanUpdateAnn.SecondPeer = ourPolicy.SecondPeer
+		chanUpdateAnn.CLTVExpiryDelta = ourPolicy.CLTVExpiryDelta
+		chanUpdateAnn.HTLCMinimumMsat = ourPolicy.HTLCMinimumMsat
+		chanUpdateAnn.HTLCMaximumMsat = ourPolicy.HTLCMaximumMsat
+		chanUpdateAnn.FeeBaseMsat = ourPolicy.FeeBaseMsat
+		chanUpdateAnn.FeeProportionalMillionths =
+			ourPolicy.FeeProportionalMillionths
+
+	case storedFwdingPolicy != nil:
+		chanUpdateAnn.FeeBaseMsat.Val = uint32(
+			storedFwdingPolicy.BaseFee,
+		)
+		chanUpdateAnn.FeeProportionalMillionths.Val = uint32(
+			storedFwdingPolicy.FeeRate,
+		)
+
+	default:
+		log.Infof("No channel forwarding policy specified for channel "+
+			"announcement of ChannelID(%v). "+
+			"Assuming default fee parameters.", chanID)
+		chanUpdateAnn.FeeBaseMsat.Val = uint32(
+			f.cfg.DefaultRoutingPolicy.BaseFee,
+		)
+		chanUpdateAnn.FeeProportionalMillionths.Val = uint32(
+			f.cfg.DefaultRoutingPolicy.FeeRate,
+		)
+	}
+
+	// With the channel update announcement constructed, we'll generate a
+	// signature that signs a double-sha digest of the announcement.
+	// This'll serve to authenticate this announcement and any other future
+	// updates we may send.
+	chanUpdateMsg, err := chanUpdateAnn.DataToSign()
+	if err != nil {
+		return nil, err
+	}
+	sig, err := f.cfg.MessageSigner.SignMessageSchnorr(
+		f.cfg.IDKeyLoc, chanUpdateMsg, false, nil,
+		chanUpdateAnn.DigestTag(),
+	)
+	if err != nil {
+		return nil, errors.Errorf("unable to generate channel "+
+			"update announcement signature: %v", err)
+	}
+	chanUpdateAnn.Signature, err = lnwire.NewSigFromSignature(sig)
+	if err != nil {
+		return nil, errors.Errorf("unable to generate channel "+
+			"update announcement signature: %v", err)
+	}
+
+	if !withProof {
+		return &chanAnnouncement{
+			chanAnn:       &chanAnn,
+			chanUpdateAnn: &chanUpdateAnn,
+		}, nil
+	}
+
+	chanAnnMsg, err := chanAnn.DigestToSign()
+	if err != nil {
+		return nil, err
+	}
+
+	pubKeys := []*btcec.PublicKey{
+		localPubKey,
+		remotePubKey,
+		localFundingKey.PubKey,
+		remoteFundingKey,
+	}
+
+	rBtc, rNode, haveReceivedNonces := f.nonceMgr.getReceivedNonces(chanID)
+	if err != nil {
+		return nil, err
+	}
+
+	if !haveReceivedNonces {
+		return nil, fmt.Errorf("havent received all the necessary " +
+			"nonces yet")
+	}
+
+	sBtc, sNode, err := f.nonceMgr.getNoncesToSend(
+		chanID, channel.RevocationProducer,
+		channel.LocalChanCfg.MultiSigKey.PubKey,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	musigBtcSess, err := f.cfg.MuSig2Signer.MuSig2CreateSession(
+		input.MuSig2Version100RC2, localFundingKey.KeyLocator,
+		pubKeys, &input.MuSig2Tweaks{},
+		[][66]byte{sNode.PubNonce, rBtc, rNode}, sBtc,
```

Comment:
Looks like the sender (us in this case), will generate unique nonces for the BTC vs the node signature, why doesn't the receiver also generate unique nonces for this instance?